<<<<<<< HEAD
From 8af7e4cbf01cce9854904869db501671c48cdced Mon Sep 17 00:00:00 2001
From: mpvader <mvader@victronenergy.com>
Date: Sun, 6 Jul 2025 22:56:46 +0200
Subject: [PATCH] package.json: change optional to mandatory and add
=======
From 6fa5c4765d8dacfe56e27eba7040cf70618cbc83 Mon Sep 17 00:00:00 2001
From: mpvader <mvader@victronenergy.com>
Date: Fri, 4 Jul 2025 21:17:28 +0200
Subject: [PATCH 2/3] package.json: change optional to mandatory and add
>>>>>>> 954c748f (signalk-server v2.14.4 + drop unused dependencies)
 signalk-venus-plugin

This changes nothing to the build, as they were already always installed. But
not anymore as per the recent change to npm-online-install.bbclass.

For the reasons why each package is wanted see here:
https://docs.google.com/document/d/1oJLH7MoRAPoqlo9aaO0gnBYA05a2ruQWFsexZfqWm9E

For why we don't want packages to be listed as optional, see here:
https://github.com/victronenergy/venus/wiki/npm-and-nodejs-packages

Part of https://github.com/victronenergy/venus/issues/1304
---
 package.json | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/package.json b/package.json
index 22ab9330..64895544 100644
--- a/package.json
+++ b/package.json
@@ -120,9 +120,7 @@
     "swagger-ui-express": "^4.5.0",
     "unzipper": "^0.10.10",
     "uuid": "^8.1.0",
-    "ws": "^7.0.0"
-  },
-  "optionalDependencies": {
+    "ws": "^7.0.0",
     "@mxtommy/kip": "^3.2.0",
     "@signalk/freeboard-sk": "^2.0.0-beta.3",
     "@signalk/instrumentpanel": "0.x",
@@ -133,7 +131,8 @@
     "mdns": "^2.5.1",
     "serialport": "^11.0.0",
     "signalk-n2kais-to-nmea0183": "^1.3.1",
-    "signalk-to-nmea2000": "^2.16.0"
+    "signalk-to-nmea2000": "^2.16.0",
+    "signalk-venus-plugin": "^2.1.0"
   },
   "devDependencies": {
     "@eslint-react/eslint-plugin": "^1.45.1",
-- 
2.34.1

